src/sentry/receivers/onboarding.py

```diff
@@ -262,15 +262,16 @@ def record_first_cron_monitor(project, user, from_upsert, **kwargs):
 
 @first_cron_checkin_received.connect(weak=False)
 def record_first_cron_checkin(project, monitor_id, **kwargs):
-    project.update(flags=F("flags").bitor(Project.flags.has_cron_checkins))
+    updated = project.update(flags=F("flags").bitor(Project.flags.has_cron_checkins))
```

Comment:
To guard against duplicates here properly, you need to update the row only when that flag isn't already set.

So you need `Project.objects.filter(<where flag not set>).update(flags=F("flags").bitor(Project.flags.has_cron_checkins))`

The where check guarantees that only 1 update will every make it through here, and helps us avoid any race conditions that could cause duplicates.

Example here: https://github.com/getsentry/sentry/blob/ab8f507e40c095906890f1261b7b649494f6b4d0/src/sentry/receivers/onboarding.py#L398-L400